File: DataStructure/217_containsDuplicate/main.cpp
#include <bits/stdc++.h>
using namespace std;
bool containsDuplicate(vector<int> &nums)
{
    map<int, int> m;
    for (int i = 0; i < nums.size(); i++)
    {
        if (m[nums[i]] > 0)
        {
            return true;
        }
        m[nums[i]] = 1;
    }
    return false;
}
int main()
{
    vector<int> a;
    a.push_back(-2);
    a.push_back(3);
    a.push_back(-4);

    cout << containsDuplicate(a) ? 1 : 0 << endl;
    return 0;
}

File: DataStructure/217_containsDuplicate/main.py
from pip import List


def containsDuplicate(self, nums: List[int]) -> bool:
    hash = {}
    for i in range(len(nums)):
        if nums[i] in hash:
            hash[nums[i]] += 1
        else:
            hash[nums[i]] = 1
        if hash[nums[i]] > 1:
            return True
    return False


nums = [2, 2, 11, 15]
ans = containsDuplicate(0, nums)
print(ans)
